### PR TITLE
Add tests for TRASH_POST effects

### PR DESCRIPTION
**Pull request by BE-Webdesign**
Add unit test coverage TRASH_POST effects.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style.
- [x] My code follows has proper inline documentation.

### CODE REVIEW

File: editor/test/effects.js [hunk 1]
```diff
@@ -321,6 +322,96 @@ describe( 'effects', () => {
...
+				id: 1,
+				title: {
+					raw: 'A History of Pork',
+				},
+				content: {
+					raw: '',
+				},
+				status: 'draft',
+			};
+			wp.api = apiCall( post );
```
Comment (gziolo) [inline editor/test/effects.js]: What do you think about stubing `wp.api.getPostTypeModel` method instead? We could verify if `getCurrentPostType` selector returns proper object from the state. This would also allow us to make  `getPostTypeModel` mock much shorter.

File: editor/test/effects.js [hunk 2]
```diff
@@ -321,6 +322,96 @@ describe( 'effects', () => {
...
+			const handler = effects.TRASH_POST;
+			const dispatch = jest.fn();
+			const store = { getState: () => ( { currentPost: post } ), dispatch };
+
+			const action = { postId: 1 };
+
+			handler( action, store );
+
+			expect( dispatch ).toHaveBeenCalledTimes( 2 );
+			expect( dispatch.mock.calls[ 0 ][ 0 ] ).toEqual( removeNotice( 'TRASH_POST_NOTICE_ID' ) );
```
Comment (gziolo) [inline editor/test/effects.js]: Is there by any chance a nicer API offered by Jest to verify if the method was called with proper values? `toHaveBeenCalledWith` could work, but it doesn't check order. Maybe that would enough though.

File: editor/test/effects.js [hunk 3]
```diff
@@ -321,6 +322,96 @@ describe( 'effects', () => {
...
+			expect( dispatch.mock.calls[ 1 ][ 0 ] ).toEqual( {
+				...action,
```
Comment (gziolo) [inline editor/test/effects.js]: It might be easier to read to see here `postId: 1,` instead.

**Comment by gziolo** [comment]
In general looks great. I would love us to make the individual smaller so it would be easier to read what's happening inside. I left some ideas to consider. Personally, I'm not happy that we have to use global object here. Anyway, since we use it, my proposal is to stub/mock it instead of overriding. This may enforce us to use a logic that resets mock after each test (not sure about it).

File: editor/test/effects.js (hunk 2 above)
Comment (BE-Webdesign) [reply editor/test/effects.js]: I like checking the exact stuff. toHaveBeenCalledWith is fine, if the array lookup is undesirable. Will change.

Comment (gziolo) [reply editor/test/effects.js]: It exposes the internal structure of the mock object at the moment, which is hard to follow in my opinion.

**Comment by BE-Webdesign** [comment]
Updated.

**Comment by gziolo** [comment]
@BE-Webdesign, there was a change introduced to the way API requests are handled. Would you mind updating this PR to reflect it? Otherwise, it should be good to go. I missed your comment ... There were too many tasks PRs open. I'm close to having everything sorted out :)
